plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -135,20 +135,26 @@ export class KubernetesProxy {
       const logger = this.logger.child({ cluster: originalCluster.name });
       middleware = createProxyMiddleware({
         logProvider: () => logger,
+        ws: true,
         secure: !originalCluster.skipTLSVerify,
         changeOrigin: true,
+        pathRewrite: { [`^${originalReq.baseUrl}`]: '' },
         router: async req => {
-          // Re-evaluate the cluster on each request, in case it has changed
-          const cluster = await this.getClusterForRequest(req);
+          const isWS = req.headers.upgrade === 'websocket';
+          const cluster = isWS
+            ? originalCluster
+            : await this.getClusterForRequest(req);
```

Comment:
This makes me a little nervous -- I imagine a scenario like this:

1. I'm using the catalog clusterLocator and I have a cluster called `cluster-a` in my catalog
1. I make some (non-WebSocket) proxy request to `cluster-a` -- this is where `originalCluster` gets initialized and effectively cached in memory
1. I change the URL of `cluster-a` in the catalog
1. I make a WebSocket request to `cluster-a`

The question is, which URL will the websocket request be proxied to -- the initial URL or the new one after updating the cluster entity in the catalog?

Reply:
I'll test a bit more to see how the WS request flows through the parent function where the `originalCluster` is initialized.

Reply:
Here's what happens when I do multiple requests that land on the same proxy, on 1st request it grabs the `originalCluster`, then creates the proxy instance & lands in the router, however after that the proxy takes care of it, meaning the `originalCluster` doesn't get reevaluated, I'm not sure what happens when you change the catalog, but depending on what that does the proxy might end with outdated cluster details specifically for WS requests.

If changing the catalog triggers a server restart then there's no issue, but if it only changes the data in memory, then it is an issue, I'm going to change this to avoid `header()` function, and instead go for the shared `headers[]` object.